katsdpcontroller/test/test_product_config.py

```diff
@@ -386,6 +396,32 @@ def test_centre_frequency_mismatch(self) -> None:
                 Options(), 'wide1_acv', self.config, self.src_streams, {}
             )
 
+    def test_input_labels(self) -> None:
+        self.config['input_labels'] = ['m900h', 'm900v', 'm901h', 'm901v']
+        acv = NgcAntennaChannelisedVoltageStream.from_config(
+            Options(), 'wide1_acv', self.config, self.src_streams, {}
+        )
+        assert_equal(acv.input_labels, self.config['input_labels'])
+
+    def test_bad_input_labels(self) -> None:
+        self.config['input_labels'] = ['m900h']
+        with assert_raises_regex(ValueError, 'input_labels has the wrong number of elements'):
```

Comment:
The change that I suggested will probably break this.

As an aside, why is this `assert_raises_regex()` function used instead of, say `@pytest.mark.xfail`?

Reply:
Firstly katsdpcontroller uses nosetests rather than pytest (because it's old - I'd like to port it one day). But I don't think this is the right place for xfail: xfail is for tests that you would really like to pass but the don't yet (maybe due to a bug you can't immediately fix). If this was pytest I'd use `pytest.raises`.

Reply:
Understood.